store/index.js

```diff
@@ -1,46 +1,56 @@
 import axios from '~/plugins/axios'
+// import * as request from 'request'
+
 export const state = () => ({
-  emailProvider: {
-    service: 'smtp',
-    username: 'username',
-    password: 'pw'
-  }
+  ticketCreatedStatus: 'pending',
+  projectTicketCreatedStatus: 'pending'
 })
 
 export const mutations = {
-  setNewEmailProvider(state, payload) {
-    state.emailProvider = payload
+  updateTicketCreatedStatus(state, payload) {
+    state.ticketCreatedStatus = payload
+  },
+  updateProjectTicketCreatedStatus(state, payload) {
+    state.projectTicketCreatedStatus = payload
   }
 }
 
 export const getters = {
-  emailProvider(state) {
-    return state.emailProvider
-  }
+  ticketCreatedStatus: state => state.ticketCreatedStatus,
+  projectTicketCreatedStatus: state => state.projectTicketCreatedStatus
 }
 
 async function sendEmail({ state, commit }, payload, path) {
   // eslint-disable-next-line
   path = path || '/email'
-
   const emailInfo = payload
   const emailProvider = state.emailProvider
-  // eslint-disable-next-line
-  // console.log('PayLoad', payload)
   if (emailProvider.username !== '' && emailProvider.password !== '') {
     try {
       // eslint-disable-next-line
       const { res } = await axios.post(path, {
         emailInfo,
         emailProvider
       })
-      // alert('Message sent successfully')
     } catch (e) {
       alert(e)
     }
   }
 }
 
+async function createTicket({ state, commit }, payload, path) {
+  const ticketInfo = payload
+  try {
+    // eslint-disable-next-line
+    const { res } = await axios.post(path, {
+      ticketInfo
+    })
+    return ('success')
+  } catch (e) {
+    return (e)
```

Comment:
What is the reason to return (e) here? Do we want to fail gracefully?

Reply:
I removed return statement.